File: arkhe_multilang/rosetta/Agent.hs
data Agent = Agent { name :: String, counter :: Int } deriving (Show)

sendMessage :: Agent -> Agent -> Agent
sendMessage sender receiver = receiver { counter = counter receiver + 1 }

main :: IO ()
main = do
    let alice = Agent "Alice" 0
        bob = Agent "Bob" 0
        bob1 = sendMessage alice bob
        bob2 = sendMessage alice bob1
    print bob2

File: arkhe_multilang/rosetta/Agent.kt
data class Agent(val name: String, var counter: Int = 0)

fun sendMessage(sender: Agent, receiver: Agent) {
    receiver.counter++
    println("${sender.name} -> ${receiver.name}: ${receiver.counter}")
}

fun main() {
    val alice = Agent("Alice")
    val bob = Agent("Bob")
    sendMessage(alice, bob)
    sendMessage(alice, bob)
}
